src/main/java/com/palantir/ri/ResourceIdentifier.java

```diff
@@ -247,26 +291,27 @@ public static ResourceIdentifier of(String service, String instance, String type
     }
 
     private static void checkServiceIsValid(String service) {
```

Comment:
Possibly. I know we need to extra checking and our own exception messages so we won't be using them.